src/app/click-handler.directive.ts

```diff
@@ -0,0 +1,31 @@
+import { Directive, HostListener, Input } from '@angular/core';
+import { ConfigService } from './config.service';
+
+@Directive({
+  selector: '[appClickHandler]'
+})
+export class ClickHandlerDirective {
```

Comment:
Should this perhaps be called EventHandlerDirective since it handles multiple events, not just click?

Reply:
It doesn't just handle all events though... it handles events that are supposed to function as a ... click... if there's a less overloaded word I'm happy to use it...
ButtonTriggerEventDirective?
It's hardcoded to be looking for the trigger event in the button behaviour config... it's not generic enough to be "Event"

Reply:
Would we want all events to come through this handler if we started to handle more?

Reply:
I see your point though. We need the word to describe the action that all these events trigger

Reply:
InteractionEventHandler?